bench.py

```diff
@@ -0,0 +1,107 @@
+    env['CLUSTER_SIZE'] = str(args.cluster_size)
+    env['BUCKET_NAME'] = args.bucket_name
+    env['STORAGE_NAME'] = args.volume_name
+    env['STORAGE_SIZE'] = '10'  # defaults to 10GB
+
+    try:
+        create_cluster(env)
+        launch_pachyderm(args, env)
+        run_benchmark(env)
+        clean_cluster(env)
+    except Failed:
+        print("something went wrong... removing the cluster...")
+        clean_cluster(env)
+
+
+def aws(args):
+    print('AWS benchmark is not currently supported')
+
+description = '''Run a Pachyderm benchmark on a cloud provider.
+
+The typical workflow looks like this:
+
+1. You write some code.  Maybe even add some new benchmarks.
+2. You `make docker-build` to build the pachd and job-shim images.
+3. You `docker tag` the images with your own namespace.  For instance, I'd tag "pachyderm/pachd:latest" with "derekchiang/pachd:latest".
+4. You push the tagged images to your own namespace.
+5. You run this script with --pachd-image, --job-shim-image, and --pachyderm-compile-image flags pointing to your own images.  For instance:
+
+    ./bench.py --pachd-image derekchiang/pachd:latest --job-shim-image derekchiang/job-shim:latest --pachyderm-compile-image derekchiang/pachyderm_compile:latest
+'''
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Run a Pachyderm benchmark on a cloud provider.', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--provider', default='GCE', choices=['GCE', 'AWS'], help='the cloud provider to run the benchmark on.')
+    parser.add_argument('--cluster-name', default='pachyderm-benchmark-{}'.format(str(uuid.uuid4())[:8]), help="the name of the cluster to run the benchmark on; one will be created.")
+    parser.add_argument('--cluster-size', default=4, type=int, help='the number of nodes to run the benchmark on.')
+    parser.add_argument('--bucket-name', default='pachyderm-benchmark-bucket-{}'.format(uuid.uuid4()), help='the GCS/S3 bucket to use with the benchmark; one will be created.')
+    parser.add_argument('--volume-name', default='pachyderm-benchmark-volume-{}'.format(uuid.uuid4()), help='the persistent volume to use with the benchmark; one will be created.')
+    parser.add_argument('--benchmark', default='.', help='a regex expression that specifies the benchmark to run; runs all benchmarks by default.')
+    parser.add_argument('--runs', default=1, type=int, help='how many times the benchmark runs.')
+    parser.add_argument('--pachd-image', default="pachyderm/pachd:latest", help='the pachd image to use.')
+    parser.add_argument('--job-shim-image', default="pachyderm/job-shim:latest", help='the job-shim image to use.')
+    parser.add_argument('--pachyderm-compile-image', default="pachyderm/pachyderm-compile:latest", help='the pachyderm-compile image to use.')
+
+    args = parser.parse_args()
+
+    print("running the benchmark with the following arguments:")
+    print(args)
```

Comment:
I like we're printing this to stdout. If we want to record results we can just pipe to a file and also have the info about how the benchmark was run.